### PR TITLE
Clean up install_triton and install_filelock in CI

### PR DESCRIPTION
**Pull request by huydhn**
After Dockerize triton in https://github.com/pytorch/pytorch/pull/95233, we can now clean up `install_triton` and `install_filelock` in the CI to improve its reliability.

### CODE REVIEW

**Review by weiwangmeta — APPROVED** [review]
Looks nice! 

I think in similar way we can explore adding torchbench dependency installation step in a docker image. cc @xuzhao9

File: .ci/pytorch/common_utils.sh [hunk 1]
```diff
@@ -149,32 +149,6 @@ function clone_pytorch_xla() {
...
-      CXX=g++-9 pip_install --user "git+https://github.com/openai/triton@${commit}#subdirectory=python"
-    elif [[ "${BUILD_ENVIRONMENT}" == *clang* ]]; then
-      # Trition needs <filesystem> which surprisingly is not available with clang-9 toolchain
-      sudo add-apt-repository -y ppa:ubuntu-toolchain-r/test
-      sudo apt-get install -y g++-9
-      CXX=g++-9 pip_install --user "git+https://github.com/openai/triton@${commit}#subdirectory=python"
-    else
-      pip_install --user "git+https://github.com/openai/triton@${commit}#subdirectory=python"
-    fi
-    pip_install --user jinja2
```
Comment (weiwangmeta) [inline .ci/pytorch/common_utils.sh:174]: Oh, is this jinja2 included?

Comment (huydhn) [reply .ci/pytorch/common_utils.sh:174]: Yup, this has already been added to Docker requirements file https://github.com/pytorch/pytorch/blob/master/.ci/docker/requirements-ci.txt#L262

Comment (weiwangmeta) [reply .ci/pytorch/common_utils.sh:174]: nice!

**Pull request closed without merging**